js/lib/ui.js

```diff
@@ -73,6 +75,11 @@ elgg.ui.toggles = function(event) {
 			$elem.toggle();
 		}
 	});
+
+	$(document).trigger('elgg.ui.toggle', [{
+		$toggler: $this,
+		$target: $target
+	}]);
```

Comment:
The big win is the callback will be bound to the toggler and could be
accessed as `this`. Keeps it consistent with normal jquery event handlers.
On Nov 28, 2015 6:02 PM, "Ismayil Khayredinov" <info@hypejunction.com>
wrote:

> Well, if you are experimenting, we might as well give it a shot. I like
> the idea of using jQuery for handling Elgg events.
> On Nov 28, 2015 5:58 PM, "Steve Clay" <notifications@github.com> wrote:
>
>> In js/lib/ui.js
>> <https://github.com/Elgg/Elgg/pull/9159#discussion_r46081928>:
>>
>> > @@ -73,6 +75,11 @@ elgg.ui.toggles = function(event) {
>> >  			$elem.toggle();
>> >  		}
>> >  	});
>> > +
>> > +	$(document).trigger('elgg.ui.toggle', [{
>> > +		$toggler: $this,
>> > +		$target: $target
>> > +	}]);
>>
>> I replaced this with trigger_hook. What's the big win of triggering on
>> the toggler element?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Elgg/Elgg/pull/9159/files#r46081928>.
>>
>